liquidator/index.js

```diff
@@ -189,69 +195,73 @@ async function run(
+      // {"type":"cryptowatch","exchange":"binance"}]}
+      priceFeedConfig: process.env.PRICE_FEED_CONFIG ? JSON.parse(process.env.PRICE_FEED_CONFIG) : null,
+      // If there is a liquidator config, add it. Else, set to null. This config contains crThreshold,liquidationDeadline,
+      // liquidationMinPrice, txnGasLimit & logOverrides. Example config:
+      // {"crThreshold":0.02,  -> Liquidate if a positions collateral falls more than this % below the min CR requirement
+      //   "liquidationDeadline":300, -> Aborts if the transaction is mined this amount of time after the last update
+      //   "liquidationMinPrice":0, -> Aborts if the amount of collateral in the position per token is below this ratio
+      //   "txnGasLimit":9000000 -> Gas limit to set for sending on-chain transactions.
+      //   "logOverrides":{"positionLiquidated":"warn"}} -> override specific events log levels.
+      liquidatorConfig: process.env.LIQUIDATOR_CONFIG ? JSON.parse(process.env.LIQUIDATOR_CONFIG) : null,
+      // If there is a LIQUIDATOR_OVERRIDE_PRICE environment variable then the liquidator will disregard the price from the
+      // price feed and preform liquidations at this override price. Use with caution as wrong input could cause invalid liquidations.
+      liquidatorOverridePrice: process.env.LIQUIDATOR_OVERRIDE_PRICE
+    };
 
-    await run(
-      Logger,
-      process.env.EMP_ADDRESS,
-      pollingDelay,
-      errorRetries,
-      errorRetriesTimeout,
-      priceFeedConfig,
-      liquidatorConfig,
-      liquidatorOverridePrice
-    );
+    // Check if the bot is being run as a node process or as a truffle process.
+    if (typeof web3 == "undefined") {
+      // Create a web3 instance. This has built in re-try on error and loads in a provided mnemonic or private key.
+      const { web3 } = require("@umaprotocol/financial-templates-lib/clients/Web3WebsocketClient");
+      await run(Logger, web3, ...Object.values(executionParameters));
+      // Else, if the web3 instance is not undefined, then the script is being run from Truffle. Use present web3 instance.
+    } else {
+      await run(Logger, web3, ...Object.values(executionParameters));
+    }
   } catch (error) {
     Logger.error({
       at: "Liquidator#index",
       message: "Liquidator configuration error🚨",
       error: typeof error === "string" ? new Error(error) : error
     });
     await waitForLogger(Logger);
-    callback(error);
-    return;
+    process.exit(1);
   }
-  callback();
 }
 
 // Attach this function to the exported function in order to allow the script to be executed through both truffle and a test runner.
 Poll.run = run;
 module.exports = Poll;
+
+// Run the poll function by default. This enables the script to be run as a node process.
+Poll()
+  .then(() => {})
+  .catch(err => {
+    console.error(err);
+    process.exit(1);
+  });
```

Comment:
A few questions about this:
1. Instead of changing the inner function to call process.exit rather than `callback()`, could we instead pass in a callback function for the node case? Something like:
```js
function nodeCallback(err) {
  if (err) {
    console.error(err);
    process.exit(1);
  } else {
    process.exit(0);
  }
}

Poll(nodeCallback)
  .then(() => {})
  .catch(nodeCallback);
```

This would allow truffle to use its expected callback setup and node to fit into that setup. Or is it done this way because when calling `callback(err)` in truffle, it doesn't produce a nonzero exit code?

2. Do you need the empty `.then()`? Or can you just do `.catch`?

3. If you call this with truffle, will this cause it to run twice in parallel? Once when truffle tries to import the file and hits this line and once when truffle calls the `Poll` function?

Reply:
I've re-factored this to conform to your pattern, PTAL. One downside with doing it like this is we need to _not_ run the `Poll(nodeCallback)` logic if the script is being imported to the tests, otherwise, on import the test utility runs the process. Please let me know if there is a better way to deal with this situation.

Reply:
I see. Was it not running it in tests the way it was written before? If so, how? I would expected the code to be roughly equivalent in that regard.

One trick to making this work is detecting if this is the main module: https://stackoverflow.com/a/6398335, which should work (although, it may be worth testing). Basically, the above could be slightly refactored to:
```js
if (require.main === module) {
  Poll(nodeCallback)
    .then(() => {})
    .catch(nodeCallback);
}
```

This will detect if the index.js file is called via `node index.js` or imported via a `require`.

Reply:
It was working at a point but I realized the version you had originally reviewed was not working. In the tests, it would execute the main function on `require`, which clearly would cause the tests to fail.

Your solution is much cleaner, thanks for that! I've implemented the changes, PTAL.

Reply:
LGTM!